### DISCUSSION TITLE
Remove whitespace from Spacemacs' paste transient state kill ring.

### DISCUSSION THREAD

**Discussion by bhw-foss** [syl20bnr/spacemacs#15455] [Ideas]
`dotspacemacs-enable-paste-transient-state` is in our dotfile template, and I find it very useful. There exists a minor annoyance that deleted whitespace with the `dd` or `x` keys appear in the kill ring. This requires an extra `C-j/C-k` to bypass the noise. I'm sure there's someone out there whose workflow depends on this kill-ring behaviour. Perhaps others will find it convenient to default to ignoring whitespace in the kill-ring. Courtesy of stackoverflow,

```
;; https://stackoverflow.com/questions/12102554/emacs-skip-whitespace-kills
(defun night/h-kill-skip-whitespace (orig-fn string &optional rest)
  (let* ((string-raw (substring-no-properties string))
         (space-p (not (string-match-p "[^ \t\n\r]" string-raw))))
    (cond
     ((not space-p)
      (apply orig-fn string rest))
     (t
      (message "skipped whitespace kill")))))

(advice-add 'kill-new :around #'night/h-kill-skip-whitespace)
```

Am I missing a built-in Emacs function that toggles whitespace off in kill rings? Would anybody else find this a useful default to have in case the user has `dotspacemacs-enable-paste-transient-state` enabled? Food for thought, happy to hear everyone's opinions.

To sum it up:
- A small quality-of-life improvement
- At a cost of possible beginner confusion due to messing with the kill-ring
- Mitigate this through less annoying means than `message`?
- This brings default behaviour more in line with editors like Word (for better or for worse?)

**Comment by bhw-foss** [comment]
https://github.com/syl20bnr/spacemacs/pull/15634
